assert/assertions_priv_test.go

```diff
@@ -0,0 +1,77 @@
+package assert
+
+import (
+	"bufio"
+	"strings"
+	"testing"
+)
+
+func Test_indentMessageLines(t *testing.T) {
+	tt := []struct {
+		name            string
+		msg             string
+		longestLabelLen int
+		expected        string
+	}{
+		{
+			name:            "single line",
+			msg:             "Hello World\n",
+			longestLabelLen: 11,
+			expected:        "Hello World",
+		},
+		{
+			name:            "multi line",
+			msg:             "Hello\nWorld\n",
+			longestLabelLen: 11,
+			expected:        "Hello\n\t            \tWorld",
+		},
+		{
+			name:            "single line - over the bufio default limit",
+			msg:             strings.Repeat("hello ", bufio.MaxScanTokenSize+10),
+			longestLabelLen: 11,
+			expected:        strings.Repeat("hello ", bufio.MaxScanTokenSize+10),
+		},
+		{
+			name:            "multi line - over the bufio default limit",
+			msg:             strings.Repeat("hello\n", bufio.MaxScanTokenSize+10),
+			longestLabelLen: 3,
+			expected: strings.TrimSpace(
+				strings.TrimPrefix(strings.Repeat("\thello\n\t    ", bufio.MaxScanTokenSize+10), "\t"),
+			),
+		},
+		{
+			name:            "single line - just under the bufio default limit",
+			msg:             strings.Repeat("hello ", bufio.MaxScanTokenSize-10),
```

Comment:
The boundary is not `"hello"` repeated `bufio.MaxScanTokenSize` times, but a string which has a line whose length is `bufio.MaxScanTokenSize` bytes.

We probably need a utility function to build such an input string. At least `strings.Repeat` alone doesn't help.

Reply:
Ohh! that's right. 

Changed the approach a little bit. The new commit generates the input as you suggested. And instead of matching against an exact output, it asserts the pattern of the output. Like the leading white spaces etc. Because for generating the expected output, we will end up rewriting the actual function logic in the test.